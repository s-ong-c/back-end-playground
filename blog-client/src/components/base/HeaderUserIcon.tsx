import * as React from 'react';
import styled from 'styled-components';

const HeaderUserIconBlock = styled.div``;
interface HeaderUserIconProps{}

const HeaderUserIcon: React.SFC<HeaderUserIconProps> = props => {
  return <HeaderUserIconBlock />;
  };

export default HeaderUserIcon;