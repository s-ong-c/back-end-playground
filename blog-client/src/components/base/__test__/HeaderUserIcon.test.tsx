import * as React from 'react';
import { render } from 'react-testing-library';
import HeaderUserIcon, { HeaderUserIconProps } from '../HeaderUserIcon';

describe('HeaderUserIcon', () => {
  const setup = (props: Partial<HeaderUserIconProps> = {}) => {
    const initialProps: HeaderUserIconProps = {
      user: {
        id: '',
        username: '',
        profile: {
          thumbnail: '',
          display_name: 'test',
        },
      },
      onClick: () => undefined,
    };
    const utils = render(<HeaderUserIcon {...initialProps} {...props} />);
    return { ...utils };
  };
  it('renders property', () => {
    setup();
  });
  it('matches snapshot', () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });
});
