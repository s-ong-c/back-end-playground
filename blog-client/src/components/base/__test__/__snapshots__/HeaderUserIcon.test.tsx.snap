// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderUserIcon matches snapshot 1`] = `
<div>
  <div
    class="sc-bdVaJa htwiTs"
  >
    <img
      alt="thumbnail"
      src="default_thumbnail.png"
    />
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
  </div>
</div>
`;
